src/shared/tests/file-record.js

```diff
@@ -0,0 +1,148 @@
+import fs from 'fs';
+import path from 'path';
+import { fileURLToPath } from 'url';
+
+import test from 'ava';
+
+import * as fileRecord from '../file-record.js';
+
+test('isDirectory', t => {
+  t.plan(6);
+  t.true(fileRecord.isDirectory({ entries: [] }));
+  t.true(fileRecord.isDirectory({ name: 'folder', entries: [] }));
+  t.true(fileRecord.isDirectory({ entries: [{ bufferData: new Uint8Array(16) }] }));
+  t.true(
+    fileRecord.isDirectory({
+      name: 'folder',
+      entries: [{ name: 'file', bufferData: new Uint8Array(16) }],
+    })
+  );
+  t.false(fileRecord.isDirectory({ bufferData: new Uint8Array(16) }));
+  t.false(fileRecord.isDirectory({ name: 'file', bufferData: new Uint8Array(16) }));
+});
+test('isFile', t => {
+  t.plan(6);
+  t.true(fileRecord.isFile({ bufferData: new Uint8Array(16) }));
+  t.true(fileRecord.isFile({ name: 'file', bufferData: new Uint8Array(16) }));
+  t.false(fileRecord.isFile({ entries: [] }));
+  t.false(fileRecord.isFile({ name: 'folder', entries: [] }));
+  t.false(fileRecord.isFile({ entries: [{ bufferData: new Uint8Array(16) }] }));
+  t.false(
+    fileRecord.isFile({
+      name: 'folder',
+      entries: [{ name: 'file', bufferData: new Uint8Array(16) }],
+    })
+  );
+});
+test('createHost', t => {
+  t.plan(10);
+  function assertImplementsHost(actualHost) {
+    t.is(typeof actualHost.read, 'function');
+    t.is(typeof actualHost.collapse, 'function');
+  }
+  assertImplementsHost(fileRecord.host);
+  assertImplementsHost(fileRecord.createHost());
+  assertImplementsHost(fileRecord.createHost({ fs: fs.promises, path: path }));
+  assertImplementsHost(fileRecord.createHost({ fs: { async readdir() {}, async readFile() {} } }));
+  assertImplementsHost(fileRecord.createHost({ path: { sep: '/', join() {} }, path: path.posix }));
```

Comment:
This object initializer has a duplicated property name. Was this meant to be two different tests?

Reply:
Un. I meant to only have the first mirroring the mock fs createHost before it.